### PR TITLE
[ᚬmaster] feat(trace): add moodyblue trace

### PR DESCRIPTION
**Pull request by KaoImin**
**What type of PR is this?**
feature

**What this PR does / why we need it**:
* add moodyblus trace in consensus module
* let bls private key extended from secp256k1 private key 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by yejiayu** [comment]
Test passed by 4 nodes?

**Comment by KaoImin** [comment]
> Test passed by 4 nodes?

Yes

File: core/consensus/src/synchronization.rs [hunk 1]
```diff
@@ -82,9 +97,25 @@ where
             self.check_proof(id, proof.clone())?;
 
             if id != 1 && current_hash != epoch.header.pre_hash {
+                trace::error(
+                    "prev_hash_diff".to_string(),
```
Comment (homura) [inline core/consensus/src/synchronization.rs]: The event is best described by an action, recommended to use `diff_prev_epoch_hash`

File: core/consensus/src/synchronization.rs [hunk 2]
```diff
@@ -82,9 +97,25 @@ where
...
+                    Some(json!({
+                        "epoch_id": id,
+                        "current_prev_hash": epoch.header.pre_hash.as_hex(),
```
Comment (homura) [inline core/consensus/src/synchronization.rs]: If this should be `current_hash.as_hex()`

Comment (KaoImin) [reply core/consensus/src/synchronization.rs]: Why? This means the `prev_hash` are different.

**Pull request merged**